quipucords/api/deployments_report/view.py

```diff
@@ -64,27 +64,41 @@ def deployments(request, pk=None):
             'report_id': [_(messages.COMMON_ID_INV)]
         }
         raise ValidationError(error)
-
+    mask_report = request.query_params.get('mask', False)
     report = get_object_or_404(DeploymentsReport.objects.all(), report_id=pk)
     if report.status != DeploymentsReport.STATUS_COMPLETE:
         return Response({'detail':
                          'Deployment report %s could not be created.'
                          '  See server logs.' % report.details_report.id},
                         status=status.HTTP_424_FAILED_DEPENDENCY)
 
-    return Response(build_cached_json_report(report))
+    return Response(build_cached_json_report(report, mask_report))
 
 
-def build_cached_json_report(report):
+def build_cached_json_report(report, mask_report):
     """Create a count report based on the fingerprints and the group.
 
     :param report: the DeploymentsReport used to group count
     :returns: json report data
     :raises: Raises validation error group_count on non-existent field.
     """
-    return {'report_id': report.id,
-            'status': report.status,
-            'report_type': report.report_type,
-            'report_version': report.report_version,
-            'report_platform_id': str(report.report_platform_id),
-            'system_fingerprints': json.loads(report.cached_fingerprints)}
+    system_fingerprints = json.loads(
+        report.cached_fingerprints)
+    if validate_query_param_bool(mask_report):
+        if report.cached_masked_fingerprints:
```

Comment:
Would it be possible to use a `get_object_or_404(report.cached_masked_fingerprints)`? I am not sure, but just wanted to throw it out there.

Reply:
Actually, after thinking about it for a bit and looking at the django [doc](https://docs.djangoproject.com/en/3.0/topics/http/shortcuts/#get-object-or-404) I don't think this possible. You would probably need some type of pk or something. I am gonna go ahead and resolve this one.